pkg/reconciler/reconciler.go

```diff
@@ -86,5 +101,22 @@ func (reconciler *Reconciler) reconcileDebounced(ctx context.Context) error {
 	if err := r.reconcileLayer2(l2vnis); err != nil {
 		return err
 	}
+
+	if !reconciler.healthChecker.IsInitialized() {
```

Comment:
Why are we doing all these checks if the healthChecker is not yet initialized?

Wouldn't we expect them to fail anyway?

Reply:
I think the method name was not descriptive enough. Changed it to `IsNetworkingHealthy()`.

The whole point of this check is, because reconciler runs in a loop, to run the healthcheck only if networking was not already checked and determined to be healthy, and set `isNetworkingHealhy` (formerly `isInitialized`) variable accordingly. After it is set, node taint should be removed and healthcheck should not be performed again.